module_utils/ansible_nailgun_cement.py

```diff
@@ -308,6 +323,16 @@ def find_content_view_filter(module, name, content_view, failsafe=False):
     return handle_find_response(module, content_view_filter.search(), message="No content view filter found for %s" % name, failsafe=failsafe)
 
 
+def find_host(module, name, failsafe=False):
+    host = Host(name=name).search(set(), {'search': 'name="{}"'.format(name)})
```

Comment:
Do we need to send `name` to nailgun twice? I would assume,`Host().search(set(), {'search': 'name="{}"'.format(name)})` would also work? I'm not sure if there is a beneift to one or the other?

Same for Hostgroup below...
 
Actually, it looks like we do it for `compute_profile`, `organization` and a couple of others as well.. Not sure why?

Reply:
That is, because searching is mostly a guessing game in nailgun. But yes, providing it once should be sufficient. Just need to figure out which one. And the answer will not be the same for all entities.

Reply:
@sean797 agreed and fixed